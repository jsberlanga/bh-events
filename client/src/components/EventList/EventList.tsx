import * as React from 'react';
import styled from 'styled-components';
import { useEvents } from 'hooks';
import { formatDate } from 'helpers';

const List = styled.ul`
  list-style: none;
  padding: 0;
  display: flex;
`;

const ListItem = styled.li`
  margin-right: 1rem;
  padding: 1rem;
  border-radius: 4px;
  border: 1px solid;

  div:last-of-type {
    margin-bottom: 1rem;
  }

  button {
    margin-right: 1rem;
  }
`;

const EventList = () => {
  const { events, isLoading, error } = useEvents();

  if (isLoading) return <div>loading...</div>;
  if (error || !events) return <div>An error ocurred</div>;

  return (
    <React.Fragment>
      <h2>List of all the events</h2>
      <List>
        {events.map((event) => {
          return (
            <ListItem key={event._id}>
              <div>
                <strong>Name:</strong> {event.name}
              </div>
              <div>
                <strong>Lastname:</strong> {event.lastname}
              </div>
              <div>
                <strong>Email:</strong> {event.email}
              </div>
              <div>
                <strong>Date:</strong> {formatDate(event.date)}
              </div>
              <button>edit</button>
              <button>delete</button>
            </ListItem>
          );
        })}
      </List>
    </React.Fragment>
  );
};

export default EventList;
