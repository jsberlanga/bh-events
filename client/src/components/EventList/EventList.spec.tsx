import { render, screen } from '@testing-library/react';
import EventList from './EventList';

jest.mock('hooks', () => ({
  useEvents: () => ({
    events: [
      {
        name: 'test_name',
        lastname: 'test_lastname',
        email: 'test@example.com',
        date: '2020/01/01',
      },
    ],
  }),
}));

describe('EventList', () => {
  it('should render with styles', () => {
    const { container } = render(<EventList />);

    expect(container).toMatchSnapshot();
  });

  it('should render events correctly', () => {
    render(<EventList />);

    expect(screen.getByText('Name:')).toBeInTheDocument();
    expect(screen.getByText('test_name')).toBeInTheDocument();
    expect(screen.getByText('Lastname:')).toBeInTheDocument();
    expect(screen.getByText('test_lastname')).toBeInTheDocument();
    expect(screen.getByText('Email:')).toBeInTheDocument();
    expect(screen.getByText('test@example.com')).toBeInTheDocument();
    expect(screen.getByText('Date:')).toBeInTheDocument();
    expect(screen.getByText('01/01/2020')).toBeInTheDocument();
  });
});
