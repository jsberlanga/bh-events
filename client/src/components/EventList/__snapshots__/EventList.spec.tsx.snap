// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList should render with styles 1`] = `
.c0 {
  list-style: none;
  padding: 0;
  display: grid;
  grid-template-columns: 1fr;
  grid-gap: 1rem;
}

.c1 {
  margin-right: 1rem;
  margin-bottom: 1rem;
  padding: 1rem;
  height: 14rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 div:last-of-type {
  margin-bottom: 1rem;
}

.c1 button {
  margin-right: 1rem;
}

@media screen and (min-width:768px) {
  .c0 {
    grid-template-columns: repeat(3,33%);
  }
}

<div>
  <h2>
    List with all recorded events
  </h2>
  <ul
    class="c0"
  >
    <li
      class="c1 withBorder"
    >
      <div>
        <div>
          <strong>
            Name:
          </strong>
           
          test_name
        </div>
        <div>
          <strong>
            Lastname:
          </strong>
           
          test_lastname
        </div>
        <div>
          <strong>
            Email:
          </strong>
           
          test@example.com
        </div>
        <div>
          <strong>
            Date:
          </strong>
           
          01/01/2020
        </div>
      </div>
      <div>
        <button
          class="btn btn--delete"
        >
          delete event
        </button>
      </div>
    </li>
  </ul>
</div>
`;
